### ISSUE TITLE
[10.0] Replenishment and Fulfillment workflow confusion

### ISSUE THREAD

**Issue by webtekindo** [odoo/odoo#18767]
I just want to take a simple use case of an eCommerce B2C company that doesn't involve any manufacturing. 

> **Replenishment**
> 
> Replenishment is the process of reordering items that you have previously purchased in order to meet demand. For example, imagine  that you ordered 100 t-shirts, but now you are low on stock.  Instead of creating a new order from scratch, you will “replenish” your  current stock levels by reordering more t-shirts. The process of replenishment moves inventory between facilities or between suppliers to meet your consumer demand.
> 
> Replenishment should occur when your products reach their reorder point. The reorder point is the time when you need replenish your inventory levels to avoid stock shortages from occurring. In an inventory management system, the reorder point is calculated by the parameters you set for each item, such as minimums, maximums, lead times, and safety stock.

**Replenishment** is totally clear in Odoo, we have to use "Buy" method, default "Make to Stock"  (no "Make to Order" route selected) and configure an Order Points. And it's working perfect!

> **Fulfillment**
> 
> Fulfillment encompasses all of the movements that take a product from the supplier to their customer. Fulfillment includes, but is not limited to: receiving, processing,  delivery, and returns of the product. Fulfillment processes are often unique to each organization and can vary in complexity.  Replenishment focuses on the quantity of inventory being restocked, whereas fulfillment deals with the processes involved in completing an order.

**Fulfillment** is still unclear to me, for example in the simple use case of a Sale Order containing product that are not currently in stock and that we don't plan to stock (but we also may have them on stock for x reasons like: customer return, or we just decide to stop stocking that product...). 

Odoo offers different options that all seems incomplete to me:

### 1. Use the "Make to Order" Route:

_Benefit:_ 
- Good traceability of the products through the supply chain.

_Downside:_ 
- It won't check for any stock available, and will always create new Procurement/Purchase Order
- Problem in specific situations where related moves may stay in "Waiting Another Move" state forever.

It's a perfect solution for product that need customization for a specific customer, so these always have to be order regardless of the stock available. But that solution cannot be directly use for Fulfillment of classic product as it doesn't look to any potential stock available.

### 2. Use the default "Make to Stock" (no "Make to Order" selected):

In version prior to Odoo 8 (if i'm not wrong), by using the "Make to Stock" route without "Reordering Rules" the procurement will be in "Exception" ("No reorder point defined") state and that allows us to change the procurement method to "Make to Order" and run the procurement again. It can be a very painful work if a company deals with a lot of backorders.

In Odoo 10, if we use only "Make to Stock" route without "Reordering Rules", absolutely nothing will happens, no "Procurement Exceptions" and no "Purchase Oder" will be generate. So customer may wait forever and we won't notice it.

An alternative solution will be to use "Make to Stock" and set a "Reordering Rules" with "Minimum Quantity" and "Maximum Quantity" set to 0. But then it should be nice to have the Procurement Exception (as Odoo 7) back in the system as @spadae22 mentions here https://github.com/odoo/odoo/issues/3137.

### 3. Use an hybrid solution like MTS + MTO:

(from OCA https://github.com/OCA/stock-logistics-warehouse/tree/10.0/stock_mts_mto_rule)

The solution seems to solve some problems and keep a good traceability over the product.

A downside in some specific situations is that we may have some incoming stock.move for the same product mixed between MTS and MTO, and at reception of products we will have to prioritize the MTO moves instead of the MTS moves, regardless of Purchase Order/Invoice where they come from (it also may interfere with the stock.pack.operation that only prioritize first move, first choice, regardless it's MTS or MTO). 

### Conslusion:

**"Make to Order"** solution seems to be the perfect choice for product that include customization for a specific customer, but not for classic product as we may have stock available.

Now my concern is which solution we should implement to handle Fulfillment of classic product that we don't have on stock or we may still have few (but we don't plan to replenish no ordering rule)

Should we use **Hybrid MTS + MTO** or **Make to Stock + Reordering Rules (Min/Max = 0)** ?

I would love to have your experience on this problem, and how you handle it, I believe that will helps a lot of Odoo users in the same situation as us.

Thank you!

**Comment by ThijsvOers** [comment]
Hi @webtekindo ,

After reading you're case I think what you are looking is not MTS or MTO but a third solution which I think is not yet available in Odoo. 

In the software company I worked before starting with Onestein we had created the following solution:
On the product card we had three options for reordering:
- 1. order on stock rules (MTS)
- 2. order per customer (MTO)
- 3. everything that is not 1 or 2. Option 3 meant that the product could always be sold and that no reordering rule was necessary (but optional). Instead there was an extra planning rule which  looked at all the Salesorders with line items with ordertype 3 (in combination with a date period and/or different warehouses) and generated the PO based on these lines (even if there where no reordering rules defined for the product). 

What made the difference from ordertype 2 was that the products where appointed to the SO's in a different process (could be done manually but also with an automated  planning rule). So the product was allocated, but not immediately appointed to the SO (that was done in an other process and depended on the available stock).

Ordertype 3 was used for products that were often sold and should easily be available. For example, we had a client which sold garden furniture. The products were bought in large quantanties in China (every week containers of new products came in). Ordetype 3 allowed the salespersons in the stores to always sell the products where the purchase department had control of the purchasing method and the fulfillment of the order was controlled by the warehouse department. 

Maybe it's an idea to discuss furthermore your question within OCA or with Odoo. I think that there are more companies that struggle with the choice they have to make between MTO and MTS. Or maybe there is already a solution and i'm not aware of it :)

Regards,
Thijs van Oers
